protocol/storage/deals_awaiting_seal_test.go

```diff
@@ -0,0 +1,198 @@
+package storage
+
+import (
+	"testing"
+
+	"github.com/filecoin-project/go-filecoin/proofs/sectorbuilder"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDealsAwaitingSealAdd(t *testing.T) {
+	tf.UnitTest(t)
+
+	newCid := types.NewCidForTestGetter()
+	cid0 := newCid()
+	cid1 := newCid()
+	cid2 := newCid()
+	commitSectorCid := newCid()
+
+	wantSectorID := uint64(42)
+	wantSector := &sectorbuilder.SealedSectorMetadata{SectorID: wantSectorID}
+	someOtherSectorID := uint64(100)
+
+	wantMessage := "boom"
+
+	t.Run("add before success", func(t *testing.T) {
+		dealsAwaitingSeal := newDealsAwaitingSeal()
+		gotCids := []cid.Cid{}
+		dealsAwaitingSeal.onSuccess = func(dealCid cid.Cid, sector *sectorbuilder.SealedSectorMetadata) {
+			assert.Equal(t, sector, wantSector)
+			gotCids = append(gotCids, dealCid)
+		}
```

Comment:
Set `onFail` too and check that it's never called. Ditto for the other tests: check both successes and failures.